test/python/WMCore_t/Storage_t/Backends_t/CPImpl_t.py

```diff
@@ -0,0 +1,62 @@
+import os
+import unittest
+
+from mock import mock, call, Mock
+
+from WMCore.Storage.Backends.CPImpl import CPImpl
+from WMCore.WMBase import getTestBase
+
+
+class CPImplTest(unittest.TestCase):
+    def setUp(self):
+        self.CPImpl = CPImpl()
+
+    def testCreateSourceName(self):
+        self.assertEqual("name", self.CPImpl.createSourceName("", "name"))
+        self.assertEqual("file:////name", self.CPImpl.createSourceName("", "file:////name"))
+
+    @mock.patch('WMCore.Storage.Backends.CPImpl.CPImpl.run')
+    def testCreateOutputDirectory_error0(self, mock_runCommand):
+        mock_runCommand.return_value = 0
+        self.CPImpl.createOutputDirectory("dir/file/test")
+        mock_runCommand.assert_called_once_with("/bin/ls dir/file > /dev/null ")
+
+    @mock.patch('WMCore.Storage.Backends.CPImpl.CPImpl.run')
+    def testCreateOutputDirectory_error0Exception(self, mock_runCommand):
+        mock_runCommand.side_effect = Exception("Im in test, yay!")
```

Comment:
When command is called It will raise Exception, I can check If Exception handling is working as expected. Basically if there was an Exception, self.run should not been called again.